src/org/github/jamm/VM.java

```diff
@@ -0,0 +1,70 @@
+package org.github.jamm;
+
+import java.lang.management.ManagementFactory;
+
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeDataSupport;
+
+/**
+ * Utility class for retrieving information from the JVM.
+ */
+final class VM
+{
+    private static final boolean DEFAULT_USE_COMPRESSED_OOPS = true;
+    private static final int DEFAULT_ALIGNEMENT_IN_BYTES = 8;
+
+    /**
+     * Returns the value of the specified VM option
+     *
+     * @param option the option name 
+     * @return the value of the specified VM option or {@code null} if the value could not be retrieved.
+     */
+    private static String getVMOption(String option) {
+        try {
+
+            MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+            ObjectName mbean = new ObjectName("com.sun.management:type=HotSpotDiagnostic");
+            CompositeDataSupport data = (CompositeDataSupport) server.invoke(mbean, "getVMOption", new Object[]{option}, new String[]{"java.lang.String"});
+            return data.get("value").toString();
+
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    /**
+     * Retrieve the object alignment in bytes from the JVM. If the alignment cannot be retrieve 
+     * the default value will be returned.
+     *
+     * @return the object alignment in bytes if it can be retrieved or the default value (8).
+     */
+    public static int getObjectAlignmentInBytes() {
+
+        String alignement = getVMOption("ObjectAlignmentInBytes");
+        return alignement == null ? DEFAULT_ALIGNEMENT_IN_BYTES : Integer.parseInt(alignement);
+    }
+
+    /**
+     * Checks if the JVM use compressed reference.
+     *  
+     * @return {{@code true} if the JVM use compressed references {@code false} otherwise. 
+     */
+    public static boolean useCompressedOops() {
+
+        String useCommpressedOops = getVMOption("UseCompressedOops");
+        return useCommpressedOops == null ? DEFAULT_USE_COMPRESSED_OOPS : Boolean.parseBoolean(useCommpressedOops);
```

Comment:
It is the default for JDK 8

Reply:
Sure, but only in case when those flags are not used, etc, no?

Reply:
Yes. If we cannot retrieve the information from the JVM using the default is probably our best guess. Of course it can be wrong but I prefer to try the most common value that simply fail.